tsdb/index/index.go

```diff
@@ -140,7 +146,8 @@ type Writer struct {
 	symbols     *Symbols
 	symbolFile  *fileutil.MmapFile
 	lastSymbol  string
-	symbolCache map[string]uint32 // From symbol to index in table.
+	symbolCache map[string]symbolCacheEntry
+	indexCache  map[string]uint32 // From symbol to index in table.
```

Comment:
Personally I don't find these two names very useful - I can't remember which is which so I have to keep looking back at the definition.  The comment on `indexCache` talks about both "symbol" and "index".

I think in this PR `symbolCache` is only used for names, whereas `indexCache` is used for names and values, so I might suggest `nameCache` and `symbolToIndex`.

Commenting which one is used when `cacheAllSymbols` is off and on would also be useful.